deploy/ray-operator/controllers/raycluster_controller.go

```diff
@@ -0,0 +1,238 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	mapset "github.com/deckarep/golang-set"
+	"github.com/go-logr/logr"
+	_ "k8s.io/api/apps/v1beta1"
+	rayiov1alpha1 "ray-operator/api/v1alpha1"
+	"ray-operator/controllers/common"
+	_ "ray-operator/controllers/common"
+	"ray-operator/utils"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("RayCluster-Controller")
+
+// Add creates a new RayCluster Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and start it when the Manager Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &RayClusterReconciler{Client: mgr.GetClient(), Scheme: mgr.GetScheme()}
+}
+
+// add creates a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
```

Comment:
It seems that you are using kubebuilder v2, do we need add Add and newReconciler here?